Don't work with wxGrid selection in resizing mode

thesiv opened the pull request:
We should not work with the selection if we currently in the resizing mode.
So check for resizing ability at first and only then for Shift modifier.

Note that this commit is best viewed with "git diff -w" to see how little
has really changed.

thesiv committed:
Process the wxGrid mouse click depending on the current mode

Simplify the code of wxGrid::DoGridCellLeftDown() and make it more
correct by using m_cursorMode instead of duplicating (not 100%
correctly) the logic used to set it in DoGridMouseMoveEvent().

If the assumption that m_cursorMode is already set by the time
DoGridCellLeftDown() is called turns out to be wrong, e.g. when wxGrid
is used on touch screens, we would need to refactor these functions to
extract the code determining m_cursorMode from the mouse location into a
separate function that would be called from both.

Note that this commit is best viewed with "git diff -w" to see how little
has really changed.

vadz commented on src/generic/grid.cpp:
```diff
@@ -4551,92 +4551,90 @@ wxGrid::DoGridCellLeftDown(wxMouseEvent& event,
         }
-    }
-    else
+        // Return to don't set `coords` cell current.
+        return;
+    } else if ( event.CmdDown() && !event.ShiftDown() )
```
I'd like to avoid returning from the middle of this function (yes, I know that there are already existing `return`s there, but they're simpler than this one), so could we rather do:
```suggestion
    }
    else
    {
        if ( event.CmdDown() && !event.ShiftDown() )
```
and keep `SetCurrentCell(coords)` inside that `else`?

vadz reviewed (commented):
This indeed looks like the right thing to do, thanks, I'd just like to rearrange the code a little before merging it.

thesiv commented:
Done. Also I realized that `MakeCellVisible(coords);` was repeated twice, so I removed it.

vadz commented:
Thinking more about this, I wonder if we should check `m_cursorMode` here instead, as we already do in `DoGridCellLeftUp()`. I think this should make the code more obviously correct: right now, it's really not clear that the old version is wrong, and the new one is correct if you look at just this function. If we had a switch on `m_cursorMode` here instead, I think it would be much more difficult to introduce a bug here.

Of course, this relies on `m_cursorMode` being set correctly by the time we get the button press event and it's not completely obvious to me if this assumption is correct, as perhaps we could get this event without any mouse move events, e.g. when emulating mouse using touch events. But I also think that this code is already broken in this case anyhow...

And if we do it like this, we should probably revert 867cc2a3ebc which also seems like a wrong fix to the problem, retrospectively: the checks added by it are already done in `DoGridMouseMoveEvent()`, so just checking `m_cursorMode` should be sufficient.

thesiv commented:
Fixed.

thesiv committed:
Remove unused wxGrid function

thesiv committed:
Fix the cursor mode switching in wxGrid mouse move event handler

Allow switching between row and column resizing modes in mouse move event
handler.

thesiv commented:
Fixed.
